packages/core/useStorage/index.ts

```diff
@@ -218,6 +215,14 @@ export function useStorage<T extends(string | number | boolean | object | null)>
   }
 
   function update(event?: StorageEvent) {
+    if (event && event.storageArea !== storage)
+      return
```

Comment:
This condition also fixes overlapping bug when using `useStorage` on `sessionStorage` and `localStorage` with the same key